src/optionals/datetime.c

```diff
@@ -129,30 +185,291 @@ static Value strftimeNative(DictuVM *vm, int argCount, Value *args) {
 }
 
 #ifdef HAS_STRPTIME
-static Value strptimeNative(DictuVM *vm, int argCount, Value *args) {
-    if (argCount != 2) {
-        runtimeError(vm, "strptime() takes 2 arguments (%d given)", argCount);
+static Value datetimeUnix(DictuVM *vm, int argCount, Value *args) {
+    if (argCount != 0) {
+        runtimeError(vm, "unix() takes 0 arguments (%d given)", argCount);
         return EMPTY_VAL;
     }
 
-    if (!IS_STRING(args[0]) || !IS_STRING(args[1])) {
-        runtimeError(vm, "strptime() arguments must be strings");
+    Datetime *datetime = AS_DATETIME(args[0]);
+
+    return NUMBER_VAL(datetime->time);
+}
+#endif
+
+ObjAbstract *newDatetimeObj(DictuVM *vm, long time, int isLocal) {
+    ObjAbstract *abstract = newAbstract(vm, freeDatetime, datetimeTypeToString);
+    push(vm, OBJ_VAL(abstract));
+
+    Datetime *datetime = createDatetime(vm);
+    datetime->time = time;
+    datetime->isLocal = isLocal;
+
+    /**
+     * Setup Datetime object methods
+     */
+    #ifdef HAS_STRPTIME
+    defineNative(vm, &abstract->values, "unix", datetimeUnix);
+    #endif
+
+    defineNative(vm, &abstract->values, "utc", datetimeUTC);
+    defineNative(vm, &abstract->values, "before", datetimeBefore);
+    defineNative(vm, &abstract->values, "after", datetimeAfter);
+    defineNative(vm, &abstract->values, "clock", datetimeClock);
+    defineNative(vm, &abstract->values, "daysSinceJanFirst", datetimeDaysSinceJan);
+    defineNative(vm, &abstract->values, "monthsSinceJan", datetimeMonthsSinceJan);
+    defineNative(vm, &abstract->values, "isDST", datetimeIsDST);
+    defineNative(vm, &abstract->values, "daysSinceSunday", datetimeDaysSinceSunday);
+    defineNative(vm, &abstract->values, "yearsSince1900", datetimeYearsSince1900);
+    defineNative(vm, &abstract->values, "secondsAfterMinute", datetimeSecondsAfterMinute);
+    defineNative(vm, &abstract->values, "minutesAfterHour", datetimeMinutesAfterHour);
+    defineNative(vm, &abstract->values, "dayOfMonth", datetimeDayOfMonth);
+    defineNative(vm, &abstract->values, "equal", datetimeEqual);
+    defineNative(vm, &abstract->values, "timezone", datetimeTimezone);
+    defineNative(vm, &abstract->values, "toString", datetimeToString);
+    defineNative(vm, &abstract->values, "format", datetimeFormat);
+
+    abstract->data = datetime;
+    pop(vm);
+
+    return abstract;
+}
+
+static Value datetimeUTC(DictuVM *vm, int argCount, Value *args) {
+    if (argCount != 0) {
+        runtimeError(vm, "utc() takes 0 arguments (%d given)", argCount);
         return EMPTY_VAL;
     }
 
-    struct tm tictoc = {0};
-    tictoc.tm_mday = 1;
-    tictoc.tm_isdst = -1;
+    Datetime *datetime = AS_DATETIME(args[0]);
 
-    char *end = strptime(AS_CSTRING(args[1]), AS_CSTRING(args[0]), &tictoc);
+    if (datetime->isLocal) {
```

Comment:
Do we want to be dealing with local in UTC? Won't this will format out to the users timezone?